### PR TITLE
Update function for booking

### PR DESCRIPTION
**Pull request by manic1990**

### CODE REVIEW

File: BookingSystem.Tests/BAL.Tests/BookingServiceTest/GetBookingTest.cs [hunk 1]
```diff
@@ -61,15 +61,15 @@ public void GetBooking_OkResult_WithListofBooking()
...
+            _mockBookingRepository.Setup(repo => repo.GetBookingDetailsAsync()).ReturnsAsync(data);
             var _bookingService = new BookingService(_mockBookingRepository.Object, _customerRepository.Object, _attributeRepository.Object,
                                                    _roomRepository.Object, _venueRepository.Object, _cardRepository.Object,
                                                    _paymentCardRepository.Object, _realtionshipRepository.Object, _addressRepository.Object,
                                                    _rateCardRepository.Object, _chargesRepository.Object,
                                                    _paymentAttemptsRepository.Object, _paymentReversalRepository.Object, _guestProfileRepository.Object, _hotelRepository.Object);
 
             // Act
-            var result =  _bookingService.GetBookingDetailsAsync();
+            var result = _bookingService.GetBookingDetailsAsync();
```
Comment (Backiaraj07) [inline BookingSystem.Tests/BAL.Tests/BookingServiceTest/GetBookingTest.cs:73]: Change method to async

File: BookingSystem.Tests/Integration.Tests/BookingRepositoryTest/GetBookingTest.cs [hunk 2]
```diff
@@ -162,7 +162,7 @@ public void GetBookingDetails_CustomerId_NotMatched_ReturnEmptyResult()
                 bookingRepository = new BookingRepository(context);
 
                 // Act
-                var result = bookingRepository.GetBookingDetails();
+                var result = bookingRepository.GetBookingDetailsAsync();
```
Comment (Backiaraj07) [inline BookingSystem.Tests/Integration.Tests/BookingRepositoryTest/GetBookingTest.cs:166]: Make a async method, All above methods,

File: BookingSystem.DAL/Repository/BookingRepository.cs [hunk 3]
```diff
@@ -185,30 +173,19 @@ public async Task<BookingDataVM> SaveBooking(BookingDataVM model)
...
-            throw new NotImplementedException();
-        }
-
-        public bool UpdateBooking(BookingDataVM booking)
-        {
-            throw new NotImplementedException();
-        }
-
-        public BookingDataVM updateBooking(BookingDataVM model)
+        public async Task<BookingDataVM> UpdateBookingAsync(BookingDataVM model)
```
Comment (Backiaraj07) [inline BookingSystem.DAL/Repository/BookingRepository.cs:176]: Pls change the update method, Refer Venue, or hotel updates.

**Review by Backiaraj07 — COMMENTED** [review]
Comments added Pls resolve it

File: BookingSystem.BAL/Service/BookingService.cs [hunk 4]
```diff
@@ -143,33 +136,19 @@ public async Task<BookingDataVM> SaveBookingAsync(BookingDataVM booking)
 
                 if (booking.ID > 0)
                 {
-                    var bookings = _bookingRepository.updateBooking(booking);
+                    var bookings = _bookingRepository.UpdateBookingAsync(booking);
```
[the remark this answers is not in the document]
Comment (manic1990) [reply BookingSystem.BAL/Service/BookingService.cs:139]: done

File: BookingSystem.Tests/BAL.Tests/BookingServiceTest/GetBookingTest.cs (hunk 1 above)
Comment (manic1990) [reply BookingSystem.Tests/BAL.Tests/BookingServiceTest/GetBookingTest.cs]: done

File: BookingSystem.DAL/Repository/BookingRepository.cs (hunk 3 above)
Comment (manic1990) [reply BookingSystem.DAL/Repository/BookingRepository.cs:177]: done

File: BookingSystem.Tests/Integration.Tests/BookingRepositoryTest/GetBookingTest.cs (hunk 2 above)
Comment (manic1990) [reply BookingSystem.Tests/Integration.Tests/BookingRepositoryTest/GetBookingTest.cs:166]: done

**Review by Backiaraj07 — APPROVED** [review]
Review completed

**Pull request merged**